fusion/php/sample.php

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/php
+<?php
+
+/**
+ * Yelp Fusion API code sample.
+ *
+ * This program demonstrates the capability of the Yelp Fusion API
+ * by using the Business Search API to query for businesses by a 
+ * search term and location, and the Business API to query additional 
+ * information about the top result from the search query.
+ * 
+ * Please refer to http://www.yelp.com/developers/v3/documentation 
+ * for the API documentation.
+ * 
+ * Sample usage of the program:
+ * `php sample.php --term="dinner" --location="San Francisco, CA"`
+ */
+
+// OAuth credential placeholders that must be filled in by users.
+// You can find them on
+// https://www.yelp.com/developers/v3/manage_app
+$CLIENT_ID = NULL;
+$CLIENT_SECRET = NULL;
+
+// Complain if credentials haven't been filled out.
+assert($CLIENT_ID, "Please supply your client_id.");
+assert($CLIENT_SECRET, "Please supply your client_secret.");
+
+// API constants, you shouldn't have to change these.
+$API_HOST = "https://api.yelp.com";
+$SEARCH_PATH = "/v3/businesses/search";
+$BUSINESS_PATH = "/v3/businesses/";  // Business ID will come after slash.
+$TOKEN_PATH = "/oauth2/token";
+$GRANT_TYPE = "client_credentials";
+
+// Defaults for our simple example.
+$DEFAULT_TERM = "dinner";
+$DEFAULT_LOCATION = "San Francisco, CA";
+$SEARCH_LIMIT = 3;
+
+/**
+ * Given a bearer token, send a GET request to the API.
+ * 
+ * @param    $host    The domain host of the API 
+ * @param    $path    The path of the APi after the domain
+ * @return   OAuth bearer token, obtained using client_id and client_secret.
+ */
+
+function obtain_bearer_token() {
+    try {
+        # Using the build-in cURL library for easiest installation.
+        # Extension library HttpRequest would also work here.
+		$curl = curl_init();
+        if (FALSE === $ch)
+            throw new Exception('Failed to initialize');
+
+        $postfields = "client_id=" . $GLOBALS['CLIENT_ID'] .
+            "&client_secret=" . $GLOBALS['CLIENT_SECRET'] .
+            "&grant_type=" . $GLOBALS['GRANT_TYPE'];
+
+		curl_setopt_array($curl, array(
+		  CURLOPT_URL => $GLOBALS['API_HOST'] . $GLOBALS['TOKEN_PATH'],
+		  CURLOPT_RETURNTRANSFER => true,
```

Comment:
there are a lot of extra params here compared to the python example so I'm wondering if they're all necessary and if they are, can we add some comments? It will at least add some context (ie: I had to dig up why encoding is set to empty string here: http://php.net/manual/en/function.curl-setopt.php) and that seems pretty necessary)